### PR TITLE
[site] Added a styled 404 page

### PR DESCRIPTION
**Pull request by HU90m**
A slightly prettier and more informative 404 page:

Screenshot from 2023-03-27 08-19-43

Parts of the index that would be common to both it and the 404 page were separated out into Hugo partials.

### CODE REVIEW

**Comment by jwnrt** [comment]
Thanks for working on this! The old `404` message wasn't very helpful

I've tried testing this locally unsuccessfully:

* `hugo serve` from `site/landing` shows me `404 page not found` at `http://localhost:1313/foo`
* `./util/site/build-docs.sh serve` shows me a longer message, maybe from the python server? See below.

```
Error response

Error code: 404

Message: File not found.

Error code explanation: HTTPStatus.NOT_FOUND - Nothing matches the given URI.
```

Am I doing something wrong here?

File: site/landing/layouts/partials/preamble.html
Comment (jwnrt) [inline site/landing/layouts/partials/preamble.html:1]: Is this the rainbow background?

I wonder why it's defined inline, maybe we should move this out to an `.svg` and use `background-image`?

**Review by jwnrt — CHANGES_REQUESTED** [review]
Code looks fine, but I haven't tested it locally yet

Just some comments on the preamble/postamble stuff

File: site/landing/layouts/partials/preamble.html
Comment (HU90m) [reply site/landing/layouts/partials/preamble.html:1]: Yeah, I thought this was strange as well.

Now that I am leaving `index.html` untouched (and not moving this into `preamble.html`), I think I will leave this be for now.

File: site/landing/layouts/partials/postamble.html
[the remark this answers is not in the document]
Comment (HU90m) [reply site/landing/layouts/partials/postamble.html:1]: This is a great suggestion. After closer inspection, turns out very little from index is needed to keep the header and footer working, so I've left `index.html` untouched and removed the preamble & postamble partials.

**Review by jwnrt — APPROVED** [review]
Good work, looks nice.

`localhost:1313/404.html` works, so I'll trust that our server redirects to that page for 404s

**Review by hcallahan-lowrisc — APPROVED** [review]
@jwnrt Yes the hosting system will just serve a page called 404.html at the root by default. So this should work when deployed. 
LGTM!

File: site/landing/layouts/partials/preamble.html
Comment (jwnrt) [reply site/landing/layouts/partials/preamble.html:1]: Update: this was actually the icons for the "transparent... high quality... flexible..." section, they're just defined at the top for some reason.

**Comment by HU90m** [comment]
> Good work, looks nice.
> 
> `localhost:1313/404.html` works, so I'll trust that our server redirects to that page for 404s

Ah sorry, I never replied to https://github.com/lowRISC/opentitan/pull/17705#issuecomment-1486591531 . As Harry has said, you are spot on.

**Review by andreaskurth — APPROVED** [review]
LGTM, thanks @HU90m!

**Pull request merged**